### PR TITLE
Move LSA missing data calculation into the model; add missing invento…

### PR DESCRIPTION
**Pull request by eanders**
…ry CoC Codes

### CODE REVIEW

File: app/models/reports/lsa/fy2019/base.rb [hunk 1]
```diff
@@ -70,39 +72,238 @@ def value_for_options options
...
+    private def sub_population_string options
       if (sub_population = options['sub_population']) && sub_population.present?
         return "; Sub Population: #{sub_population.humanize.titleize}"
       end
       ''
     end
 
+    def missing_data(user)
+      @missing_data = {}
+      @range = ::Filters::DateRange.new(start: Date.current - 3.years, end: Date.current)
```
Comment (awisema) [inline app/models/reports/lsa/fy2019/base.rb:108]: Maybe report range and back 3 years?

File: app/models/reports/lsa/fy2019/base.rb [hunk 2]
```diff
@@ -70,39 +72,238 @@ def value_for_options options
...
+        project_name: p_t[:ProjectName].to_sql,
+        org_name: o_t[:OrganizationName].to_sql,
+        project_type: p_t[:computed_project_type].to_sql,
+        funder: f_t[:Funder].to_sql,
+        id: p_t[:id].to_sql,
+        ds_id: p_t[:data_source_id].to_sql,
+      }
+    end
+
+    private def add_missing_housing_types(user)
```
Comment (awisema) [inline app/models/reports/lsa/fy2019/base.rb]: I'd take the `add` out of these names, these don't actually add anything?
Also, I don't know if it worth it, but except for the query, are these all identical? Maybe DRY them up?

File: app/views/reports/lsa/fy2019/_form.haml [hunk 3]
```diff
@@ -9,14 +9,15 @@
           %th Project Type
           %th Operating Start Date
           %th HousingType
-          %th CoC Code
+          %th Project CoC CoC Code
```
Comment (awisema) [inline app/views/reports/lsa/fy2019/_form.haml]: This reads oddly, 'Project-CoC CoC Code'?

**Review by awisema — CHANGES_REQUESTED** [review]

File: app/models/reports/lsa/fy2019/base.rb (hunk 1 above)
Comment (eanders) [reply app/models/reports/lsa/fy2019/base.rb:108]: This gets run before the report is queued, so there's no date range available.  Checking for enrollments in the past 3 years should be sufficient.

**Pull request merged**